Source/DafnyStandardLibraries/src/DafnyStdLibs_Concurrent.py

```diff
@@ -0,0 +1,86 @@
+import threading
+from typing import Callable, Any, TypeVar, NamedTuple
+from math import floor
+from itertools import count
+
+import _dafny
+import DafnyStdLibs_Wrappers
+
+# Module: DafnyStdLibs_Concurrent
+
+
+class MutableMap:
+    def __init__(self) -> None:
+        self.map = dict()
+        self.lock = Lock()
+
+    def Keys(self):
+        self.lock.Lock__()
+        s = self.map.keys()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def HasKey(self, k):
+        self.lock.Lock__()
+        b = k in self.map
+        self.lock.Unlock()
+        return b
+
+    def Values(self):
+        self.lock.Lock__()
+        s = self.map.values()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def Items(self):
+        self.lock.Lock__()
+        s = self.map.items()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def Put(self, k, v):
+        self.lock.Lock__()
+        self.map[k] = v
+        self.lock.Unlock()
+
+    def Get(self, k):
+        self.lock.Lock__()
+        try:
+            v = self.map.get(k)
+        except KeyError:
+            self.lock.Unlock()
+            return DafnyStdLibs_Wrappers.Option_None()
+        self.lock.Unlock()
+        return DafnyStdLibs_Wrappers.Option_Some(v)
+
+    def Remove(self, k):
+        self.lock.Lock__()
+        self.map.pop(k, None)
+        self.lock.Unlock()
+
+    def Cardinality(self):
+        self.lock.Lock__()
+        l = len(self.map)
+        self.lock.Unlock()
+        return l
+
+
+class AtomicBox:
+    def __init__(self, t) -> None:
+        self.boxed = t
+
+    def Get(self):
+        return self.boxed
+        
+    def Put(self, t):
+        self.boxed = t
+
+class Lock:
+    def __init__(self) -> None:
+        self.lock = threading.Lock()
```

Comment:
How much overhead does this incur? It's good to have an actual lock by default, but there will also be lots of use cases where it's safe to assume Python is only single-threaded.

We can leave this out of scope for this PR, but I think it would be great to have some kind of configuration for multi-target Dafny projects to say "when translating to Python, make the concurrency datatypes no-ops" (and probably for other languages too)